copy/copy.go

```diff
@@ -181,10 +182,18 @@ func Image(ctx context.Context, policyContext *signature.PolicyContext, destRef,
 		// FIXME? The cache is used for sources and destinations equally, but we only have a SourceCtx and DestinationCtx.
 		// For now, use DestinationCtx (because blob reuse changes the behavior of the destination side more); eventually
 		// we might want to add a separate CommonCtx — or would that be too confusing?
-		blobInfoCache: blobinfocache.DefaultCache(options.DestinationCtx),
-
-		compressionFormat: *options.DestinationCtx.CompressionFormat,
-		compressionLevel:  options.DestinationCtx.CompressionLevel,
+		blobInfoCache:    blobinfocache.DefaultCache(options.DestinationCtx),
+		compressionLevel: options.DestinationCtx.CompressionLevel,
```

Comment:
`options.DestinationCtx` can be `nil`, per the common semantics of `*types.SystemContext`.

Reply:
@giuseppe, can you take care of the remaining clean ups for that? And line 189?